### PR TITLE
refactor: Refactor SOF, update entity ID format, update entity managment logic, update system-to-class association code, full unit tests, code clean-up, clear comments

### PR DESCRIPTION
**Pull request by fc-tezza**
SOF refactor 
 - proposed `npm `package update to v0.1.0 (package is set to `private` for now, until we are ready for integration and release)
  - Added explicit `Id` types for Entity IDs
  - reworked Entity management for simpler Class-to-Object instantiation and inheritance
  - reworked System-to-Class association and scoped enforcement to use a simpler Tagging pattern
  - update MUD Framework dependencies to @latticexyz@2.0.12
  - removed world.initialMsgSender() - full MUD execution context coming in a future PR
  - removed Hooks - a reworked hook version coming in a future PR

### CODE REVIEW

File: mud-contracts/smart-object-framework-v2/package.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
+  "name": "@eveworld/smart-object-framework",
```
Comment (0xxlegolas) [inline mud-contracts/smart-object-framework-v2/package.json]:
```suggestion
  "name": "@eveworld/smart-object-framework-v2",
```

File: mud-contracts/smart-object-framework-v2/src/interfaces/IEntities.sol [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+import { ResourceId } from "@latticexyz/world/src/WorldResourceId.sol";
+import { Id } from "../libs/Id.sol";
+
+/**
+ * @title IEntities
+ * @dev An interface for the Entities System functionality
+ */
+interface IEntities {
+	function registerClass(Id classId, Id[] memory systemTagIds) external;
```
Comment (0xxlegolas) [inline mud-contracts/smart-object-framework-v2/src/interfaces/IEntities.sol]: missing linting

**Review by 0xxlegolas — COMMENTED** [review]
LGTM! except for the build issues and linting

File: mud-contracts/smart-object-framework-v2/src/interfaces/IEntities.sol (hunk 2 above)
Comment (0xxlegolas) [reply mud-contracts/smart-object-framework-v2/src/interfaces/IEntities.sol]: looks good in local. Resolving it.

File: mud-contracts/smart-object-framework-v2/src/interfaces/ITags.sol [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+import { ResourceId } from "@latticexyz/world/src/WorldResourceId.sol";
+import { Id } from "../libs/Id.sol";
+
+/**
+ * @title ITags
+ * @dev An interface for the Tags System functionality
+ */
+interface ITags {
+	function setSystemTag(Id classId, Id systemTagId) external;
```
Comment (tinkrtailor) [inline mud-contracts/smart-object-framework-v2/src/interfaces/ITags.sol]: Nitpick: Strange indent

Comment (0xxlegolas) [reply mud-contracts/smart-object-framework-v2/src/interfaces/ITags.sol]: even I found this strange behaviour, but apparently its only in github UI. Local it should be fine

File: mud-contracts/smart-object-framework-v2/package.json (hunk 1 above)
Comment (fc-tezza) [reply mud-contracts/smart-object-framework-v2/package.json]: Done in commit 2350b72266add31123d779ce4c427a67882f4ab8

File: mud-contracts/smart-object-framework-v2/src/interfaces/ITags.sol (hunk 3 above)
Comment (fc-tezza) [reply mud-contracts/smart-object-framework-v2/src/interfaces/ITags.sol]: Linting done in  commit 2350b72266add31123d779ce4c427a67882f4ab8

**Comment by ccp-tezza** [comment]
> LGTM! except for the build issues and linting

Build issues due to `pnpm-lock.yaml `conflict.. all resolved in the most recent commit.

I've also merged with the latest develop changes

**Review by 0xxlegolas — APPROVED** [review]

**Pull request merged**